drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/ScenarioValidationService.java

```diff
@@ -0,0 +1,244 @@
+        for (FactMapping factMapping : simulation.getSimulationDescriptor().getFactMappings()) {
+            if (isToSkip(factMapping)) {
+                continue;
+            }
+
+            // try to navigate using all the steps to verify if structure is still valid
+            List<String> steps = expressionElementToString(factMapping);
+
+            try {
+                String instanceClassName = factMapping.getFactIdentifier().getClassName();
+
+                if (steps.isEmpty()) {
+                    // in case of top level simple types just try to load the class
+                    loadClass(instanceClassName, kieContainer.getClassLoader());
+                } else {
+                    Object bean = beanInstanceMap.computeIfAbsent(
+                            instanceClassName,
+                            className -> fillBean(className, Collections.emptyMap(), kieContainer.getClassLoader()));
+
+                    List<String> stepsToField = steps.subList(0, steps.size() - 1);
+                    String lastStep = steps.get(steps.size() - 1);
+
+                    ScenarioBeanWrapper<?> beanBeforeLastStep = navigateToObject(bean, stepsToField, true);
+
+                    ScenarioBeanWrapper<?> beanWrapper = navigateToObject(beanBeforeLastStep.getBean(), Collections.singletonList(lastStep), false);
+
+                    String targetClassName = beanWrapper.getBeanClass() != null ?
+                            beanWrapper.getBeanClass().getCanonicalName() :
+                            null;
+
+                    // check if target field has valid type
+                    if (!Objects.equals(factMapping.getClassName(), targetClassName)) {
+                        errors.add(createFieldChangedError(factMapping, targetClassName));
+                    }
+                }
+            } catch (ScenarioException e) {
+                errors.add(createGenericError(factMapping, e.getMessage()));
+            }
+        }
+        return errors;
+    }
+
+    protected DMNModel getDMNModel(KieContainer kieContainer, String dmnPath) {
+        return extractDMNModel(extractDMNRuntime(kieContainer), dmnPath);
+    }
+
+    private List<String> expressionElementToString(FactMapping factMapping) {
+        return factMapping.getExpressionElementsWithoutClass().stream()
+                .map(ExpressionElement::getStep).collect(Collectors.toList());
+    }
+
+    /**
+     * Skip descriptive columns (FactMappingType.OTHER), column with no instance (FactIdentifier.EMPTY)
+     * and with not expression elements
+     * @param factMapping
+     * @return
+     */
+    private boolean isToSkip(FactMapping factMapping) {
+        return OTHER.equals(factMapping.getExpressionIdentifier().getType()) ||
+                EMPTY.equals(factMapping.getFactIdentifier()) ||
+                factMapping.getExpressionElements().size() == 0;
+    }
+
+    private DMNType navigateDMNType(DMNType rootType, List<String> steps) {
```

Comment:
@danielezonca 
I think it would be better to move this and the following method to a "utility" class - DMNTypeUtils for the moment being has only one method, but if it would be refactored as a common "DMNUtil", it would be the best place to put all this DMN-related helper methods

Reply:
Done